src/table.rs

```diff
@@ -218,64 +404,74 @@ pub struct Block {
     checksum_len: usize,
 }
 
-/*
 impl Block {
     fn size(&self) -> u64 {
-        3 * mem::size_of::<usize>() as u64 + self.data.len() as u64 + self.checksum.len() as u64 + self.entry_offsets.len() as u64 * mem::size_of::<u32>() as u64
+        3 * std::mem::size_of::<usize>() as u64
+            + self.data.len() as u64
+            + self.checksum.len() as u64
+            + self.entry_offsets.len() as u64 * std::mem::size_of::<u32>() as u64
     }
 
     fn verify_checksum(&self) -> Result<()> {
-        let mut chksum = CheckSum::default();
-        chksum.merge_from_bytes(&self.checksum)?;
+        let chksum = prost::Message::decode(self.data.clone())?;
```

Comment:
I think it should be `self.checksum`.

Reply:
Yes, I forget to cherry-pick https://github.com/tikv/agatedb/pull/18/ to this branch...